x-pack/platform/plugins/shared/onechat/server/services/tools/builtin/converter.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ToolType } from '@kbn/onechat-common';
+import type { BuiltinToolDefinition, StaticToolRegistration } from '@kbn/onechat-server/tools';
+import type { InternalToolDefinition } from '../tool_provider';
+import type {
+  ToolTypeDefinition,
+  BuiltinToolTypeDefinition,
+  ToolDynamicPropsContext,
+} from '../tool_types/definitions';
+import { isBuiltinDefinition } from '../tool_types/definitions';
+
+export const convertTool = ({
+  tool,
+  definition,
+  context,
+}: {
+  tool: StaticToolRegistration;
+  definition: ToolTypeDefinition | BuiltinToolTypeDefinition;
+  context: ToolDynamicPropsContext;
+}): InternalToolDefinition => {
+  if (isBuiltinToolRegistration(tool)) {
+    return {
+      id: tool.id,
+      type: ToolType.builtin,
+      description: tool.description,
+      tags: tool.tags,
+      configuration: {},
+      readonly: true,
+      getSchema: () => tool.schema,
+      getHandler: () => tool.handler,
+    };
+  }
+  if (!isBuiltinDefinition(definition)) {
+    const getDynamicProps = () => {
+      return definition.getDynamicProps(tool.configuration, context);
+    };
+
+    return {
+      id: tool.id,
+      // static tools are surfaced as builtin
+      type: ToolType.builtin,
+      // static tools have their configuration obfuscated
+      configuration: {},
```

Comment:
The type of static tools is an implementation detail - for the end users they appear all as being `built-in`.